src/functions/update.js

```diff
@@ -0,0 +1,176 @@
+const mapServiceUrl =
+  'https://mapserv.utah.gov/arcgis/rest/services/DEQEnviro/MapService/MapServer';
+
+async function auth() {
+  console.log('initializing authentication...');
+
+  const auth = new google.auth.GoogleAuth({
+    scopes: [
+      'https://www.googleapis.com/auth/spreadsheets.readonly',
+      'https://www.googleapis.com/auth/firebase.remoteconfig',
+    ],
+  });
+
+  const authClient = await auth.getClient();
+  google.options({ auth: authClient });
+
+  return authClient;
+}
+
+export function arraysToObjects(arrays, skipFields = []) {
+  const [keys, ...values] = arrays;
+  return values.map((row) => {
+    return row.reduce((obj, value, index) => {
+      if (!skipFields.includes(keys[index])) {
+        obj[keys[index]] = value;
+      }
+
+      return obj;
+    }, {});
+  });
+}
+
+export function addFeatureServices(configs, featureServices, nameField) {
+  return configs.map((config) => {
+    const featureService = featureServices[config[nameField].split('.')[2]];
+
+    config.featureService = featureService;
+
+    return config;
+  });
+}
+
+async function getFeatureServiceLayers(propName) {
+  console.log('fetching feature service layers...');
+  const response = await got.get(`${mapServiceUrl}?f=json`).json();
+
+  return response[propName].reduce((obj, layer) => {
+    obj[layer.name] = `${mapServiceUrl}/${layer.id}`;
+
+    return obj;
+  }, {});
+}
+
+async function getConfigs(range, skipFields, featureServiceProp, nameField) {
+  const response = await sheets.spreadsheets.values.get({
+    spreadsheetId,
+    range,
+  });
+
+  let configs = arraysToObjects(response.data.values, skipFields);
+
+  configs = addFeatureServices(
+    configs,
+    await getFeatureServiceLayers(featureServiceProp),
+    nameField
+  );
+
+  return configs;
+}
+
+export function getLinksConfig(values) {
+  const [keys, ...rows] = values;
+
+  return rows.reduce((obj, row) => {
+    const id = row[keys.indexOf(fieldNames.links.id)];
+    obj[id] = {
+      description: row[keys.indexOf(fieldNames.links.description)],
+      url: row[keys.indexOf(fieldNames.links.url)],
+    };
+
+    return obj;
+  }, {});
+}
+
+async function updateRemoteConfigs(queryLayers, relatedTables, links) {
+  const remoteConfig = admin.remoteConfig();
+
+  console.log('fetching template');
+  const template = await remoteConfig.getTemplate();
+
+  if (
+    template.parameters.queryLayers.defaultValue.value === queryLayers &&
+    template.parameters.relatedTables.defaultValue.value === relatedTables &&
+    template.parameters.links.defaultValue.value === links
+  ) {
+    console.log('no changes detected, skipping remote config update');
+    return;
+  }
+
+  template.parameters.queryLayers.defaultValue.value = queryLayers;
+  template.parameters.relatedTables.defaultValue.value = relatedTables;
+  template.parameters.links.defaultValue.value = links;
+
+  console.log('validating new template');
+  await remoteConfig.validateTemplate(template);
```

Comment:
**question**: does this throw if it's not valid?

Reply:
Yes and provides a nice error message about what's going on.